tools/manifest-generator/manifest-generator.go

```diff
@@ -166,6 +165,33 @@ func fixResourceString(in string, indention int) string {
 	return out.String()
 }
 
+func evalOlmCsvUpdateVersion(inFile, csvVersion, bundleOutDir, quayNamespace, quayRepository string) (string, error) {
+	replacesCsvVersion := ""
+	if strings.Contains(inFile, ".csv.yaml") && bundleOutDir != "" {
+		bundleHelper, err := helper.NewBundleHelper("cdi", quayNamespace)
+
+		if err != nil {
+			klog.Fatalf("Failed to access quay namespace %s, repo %s, %v\n", quayNamespace, quayRepository, err)
+		}
+		if !bundleHelper.VerifyNotPublishedCSVVersion(csvVersion) {
+			klog.Fatalf("CSV version %s is already published!", csvVersion)
+		}
+		latestVersion := bundleHelper.GetLatestPublishedCSVVersion()
+		if latestVersion != "" {
+			// prevent generating the same version again
+			if strings.HasSuffix(latestVersion, csvVersion) {
+				klog.Fatalf("CSV version %s is already published!", csvVersion)
```

Comment:
Not sure how I feel about the fact that `make manifests` fatals here.  Would it be better to maybe only do this check when `olm-push`ing?

Reply:
So, which version to generate in such case:
We are extracting latest version to know what to put in replaces field in CSV manifest.
We may decide that if we are generating version that is equal to existing one then do not set anything in replaces field. We will not be able to push that manifest though since version tag is verified by quay. 
Manifest fatals when input is incorrect - why not to fatal in such case?

Reply:
My issue is basically that we are generating the olm manifests all the time, even if they are not needed/referenced.  So it may make sense to separate them from `make manifests` and have a `make olm-manifests` target that creates them.  That is not something we have to address in this PR but may be worth further discussion.